gfx_scene.cpp

```diff
@@ -1364,131 +1381,182 @@ class GfxSceneInternal
-            if(it != images.end())
+            bool has_split_extension = false;
+            if(gltf_material.extensions_count != 0)
             {
-                std::map<cgltf_image const *, std::pair<GfxConstRef<GfxImage>, GfxConstRef<GfxImage>>>::const_iterator const it2 =
-                    maps.find(it->first);
-                if(it2 != maps.end())
+                // AMD Extension support for split metal rough
+                for(uint64_t ext = 0; ext < gltf_material.extensions_count; ++ext)
                 {
-                    material_ref->roughness_map = (*it2).second.first;
-                    material_ref->metallicity_map = (*it2).second.second;
+                    if(strcmp(gltf_material.extensions[ext].name, "AMD_split_roughnessMetallic") == 0)
+                    {
+                        // Need to get the split files out of the extension
+                        std::string json = gltf_material.extensions[ext].data;
+                        json.erase(
+                            remove_if(json.begin(), json.end(),
+                                [](char c) { return (c == '\r' || c == '\t' || c == ' ' || c == '\n'); }),
+                            json.end());
+                        auto get_texture = [&](std::string_view const tag, GfxConstRef<GfxImage> &image) {
+                            if(auto pos = json.find(tag); pos != std::string::npos)
+                            {
+                                pos += tag.length();
+                                std::string clipped = json.substr(pos, json.find_first_not_of("0123456789", pos + 1) - pos);
+                                const auto index = stoi(clipped);
+                                if(index < gltf_model->textures_count)
+                                {
+                                    cgltf_texture const *split_text = &gltf_model->textures[index];
+                                    it = (split_text != nullptr ? textures.find(split_text) : textures.end());
+                                    if(it != textures.end())
+                                    {
+                                        GfxImage *temp = gfxSceneGetObject<GfxImage>(scene, (*it).second);
+                                        temp->format = ConvertImageFormatLinear(temp->format);
+                                        image = (*it).second;
+                                    }
+                                }
+                            }
+                        };
+                        constexpr std::string_view metalTag = "\"metallicTexture\":{\"index\":";
+                        constexpr std::string_view roughTag = "\"roughnessTexture\":{\"index\":";
+                        get_texture(metalTag, material_ref->roughness_map);
+                        get_texture(roughTag, material_ref->metallicity_map);
+                        has_split_extension = true;
+                    }
```

Comment:
Sounds good!
Arguably, we could also expose a `GfxMaterialFlag` along the lines of `kGfxMaterialFlag_IsReflectanceCombined` or something and handle this accordingly at the app level (then we don't diverge from the core glTF specs iiuc).
But looking good otherwise.

Reply:
We already diverged in that we have always been using split metal and roughness textures. Previously though we looked for them based on the name of the stock gltf combined metal/rough and failed if we couldnt find them. This extension basically allows for these files to be explicitly set without having the old behaviour of guessing the filename